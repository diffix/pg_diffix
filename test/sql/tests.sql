CREATE EXTENSION IF NOT EXISTS pg_diffix;
LOAD 'pg_diffix';

SET pg_diffix.default_access_level = 'publish';

CREATE TABLE test_customers (id INTEGER PRIMARY KEY, city TEXT);
INSERT INTO test_customers VALUES
  (0, NULL), (1, 'Berlin'), (2, 'Berlin'), (3, 'Rome'), (4, 'London'), (5, 'Berlin'), (6, 'Rome'),
  (7, 'Rome'), (8, 'Berlin'), (9, 'Rome'), (10, 'Berlin'), (11, 'Rome'), (12, 'Rome'), (13, 'Rome');

INSERT INTO diffix_config (rel_namespace_name, rel_name, aid_attname) VALUES
  ('public', 'test_customers', 'id');

SELECT COUNT(*) FROM test_customers;

SELECT COUNT(city) FROM test_customers;

SELECT city, COUNT(DISTINCT id) FROM test_customers GROUP BY 1;

-- Gets rejected because `city` is not the AID.
SELECT COUNT(DISTINCT city) FROM test_customers;

SELECT city FROM test_customers;

SELECT city FROM test_customers GROUP BY 1 HAVING length(city) <> 4;

SELECT COUNT(*) FROM test_customers WHERE city = 'London';

-- Get rejected because aggregators are unsupported.
SELECT SUM(id) FROM test_customers;
SELECT MIN(id) + MAX(id) FROM test_customers;
SELECT city FROM test_customers GROUP BY 1 ORDER BY AVG(LENGTH(city));
