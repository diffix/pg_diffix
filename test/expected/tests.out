CREATE TABLE test_customers (id INTEGER PRIMARY KEY, city TEXT);
INSERT INTO test_customers VALUES
 (0, NULL), (1, 'Berlin'), (2, 'Berlin'), (3, 'Rome'), (4, 'London'), (5, 'Berlin'), (6, 'Rome'),
 (7, 'Rome'), (8, 'Berlin'), (9, 'Rome'), (10, 'Berlin'), (11, 'Rome'), (12, 'Rome'), (13, 'Rome');
CREATE EXTENSION IF NOT EXISTS pg_diffix;
LOAD 'pg_diffix';
SELECT COUNT(*) FROM test_customers;
 count 
-------
    12
(1 row)

SELECT DIFFIX_COUNT(city) FROM test_customers;
 diffix_count 
--------------
            9
(1 row)

SELECT city, DIFFIX_COUNT(DISTINCT id) FROM test_customers GROUP BY 1 HAVING DIFFIX_LCF(id);
  city  | diffix_count 
--------+--------------
 Berlin |            4
 Rome   |            6
(2 rows)

SELECT city FROM test_customers;
  city  
--------
 Rome
 Rome
 Rome
 Rome
 Rome
 Rome
 Rome
 Berlin
 Berlin
 Berlin
 Berlin
 Berlin
(12 rows)

SELECT city FROM test_customers GROUP BY 1 HAVING length(city) <> 4;
  city  
--------
 Berlin
(1 row)

SELECT COUNT(*) FROM test_customers WHERE city = 'London';
 count 
-------
     0
(1 row)

